### PR TITLE
kbcmf: build on 32-bit platforms

### PR DESCRIPTION
**Pull request by maxtaco**
- add a test to see what happens if the top bit of a groupID isn't set

### CODE REVIEW

File: go/kbcmf/decrypt.go [hunk 1]
```diff
@@ -232,9 +232,12 @@ func (ds *decryptStream) processEncryptionHeader(hdr *EncryptionHeader) error {
 }
 
 func (ds *decryptStream) checkMAC(bl *EncryptionBlock, b []byte) error {
+	if (ds.keys.GroupID & groupIDMask) != groupIDMask {
+		return ErrBadGroupID(ds.keys.GroupID)
+	}
 	gid := (ds.keys.GroupID ^ groupIDMask)
 
-	if len(bl.MACs) <= gid {
+	if int64(len(bl.MACs)) <= int64(gid) {
```
Comment (zanderz) [inline go/kbcmf/decrypt.go]: Is there any chance of a high bit being extended as a sign bit here? In other words, should these be uint64?

Comment (maxtaco) [inline go/kbcmf/decrypt.go]: i don't think there's a way to get the signed big set from the above XOR.   We can also check that it's >= 0 to be sure.

**Comment by zanderz** [comment]
lgtm! :thumbsup:

**Comment by maxtaco** [comment]
thanks steve, I'll put that sanity check in.

**Pull request merged**